configure

```diff
@@ -67,6 +67,7 @@ check_toolchain()
     : ${EMACS=emacs}
     : ${BPFTOOL=bpftool}
     : ${ARCH_INCLUDES=}
+    : ${ARCH_NAME=$($CC -print-multiarch)}
```

Comment:
This needs to silence stderr, otherwise we get output like this:

```
$ CC=clang ./configure
clang-16: error: unsupported option '-print-multiarch'
clang-16: error: no input files
Found clang binary 'clang' with version 16 (from 'clang version 16.0.6')
using emacs: GNU Emacs 29.1
using bpftool v7.2.0
libbpf support: system v1.2.2
  perf_buffer__consume support: yes
  btf__load_from_kernel_by_id support: yes
  btf__type_cnt support: yes
  bpf_object__next_map support: yes
  bpf_object__next_program support: yes
  bpf_program__insn_cnt support: yes
  bpf_program__type support: yes
  bpf_program__flags support: yes
  bpf_program__expected_attach_type support: yes
  bpf_map_create support: yes
  perf_buffer__new_raw support: yes
  bpf_xdp_attach support: yes
  bpf_map__set_autocreate support: yes
  bpf_prog_test_run_opts support: yes
  bpf_xdp_query support: yes
secure_getenv support: yes
cap-ng support: yes
```

Also, probably better to put this below the check for whether `$CC` actually exists, below...